### ISSUE TITLE
Memory leak in logcollector

### ISSUE THREAD

**Issue by Molter73** [wazuh/wazuh#5013]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.13.0 | Logcollector | Manager/Agent | Packages/Sources | Linux |

## Description
A memory leak was reported on logcollector by the automated analysis on one of my PRs.

```
==15487== 4,679 bytes in 31 blocks are definitely lost in loss record 43 of 46
==15487==    at 0x4C2FB0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==15487==    by 0x61BC9B9: strdup (strdup.c:42)
==15487==    by 0x287B96: log_builder_build (log_builder.c:103)
==15487==    by 0x24B63F: SendMSGtoSCK (mq_op.c:115)
==15487==    by 0x17ADBE: w_output_thread (logcollector.c:1817)
==15487==    by 0x5F076DA: start_thread (pthread_create.c:463)
==15487==    by 0x624088E: clone (clone.S:95)
==15487== 
==15487== LEAK SUMMARY:
==15487==    definitely lost: 4,679 bytes in 31 blocks
==15487==    indirectly lost: 0 bytes in 0 blocks
==15487==      possibly lost: 1,904 bytes in 7 blocks
==15487==    still reachable: 30,559 bytes in 125 blocks
==15487==         suppressed: 0 bytes in 0 blocks
==15487== Reachable blocks (those to which a pointer was found) are not shown.
==15487== To see them, rerun with: --leak-check=full --show-leak-kinds=all
```
The problem seems to araise in the following function:
https://github.com/wazuh/wazuh/blob/1f4cc0a9e010fc6137dc1ee91fa6980054e2307d/src/shared/mq_op.c#L107-L199

When `target->log_socket->name` is `agent` and the call to `SendMSG` is successful, there is no free before the last return condition.

**Comment by vikman90** [comment]
@Molter73 this issue was fixed by #4968 in branch 3.12 (v3.12.3).

I've just merged 3.12 into 3.13 so this bug should disappear.

Please try again, and consider closing this issue.
Cheers!

**Comment by Molter73** [comment]
Just tryied it and it's fixed, thanks!
